dpdispatcher/submission.py

```
@@ -343,8 +343,37 @@ def run_submission(
+        """Report all terminal failures after other jobs and downloads finish."""
+        failed_jobs = self.failed_jobs()
+        if not failed_jobs:
+            return
+        # ``clean_jobs`` removes the local record along with the remote workdir.
+        # Re-create it before raising so callers can still inspect diagnostics
+        # and use ``dpdisp submission`` to download logs or retry the submission.
+        # Failure persistence is best effort: never hide the original job error
+        # if a custom/incomplete submission object cannot be serialized.
+        try:
+            record.write(self)
+        except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)